diff --git a/debian/bin/gencontrol.py b/debian/bin/gencontrol.py
index 1f758f300..f92066a03 100755
--- a/debian/bin/gencontrol.py
+++ b/debian/bin/gencontrol.py
@@ -684,8 +684,7 @@
         self.abiname_part = '-%s' % self.config['abi', ]['abiname']
         # We need to keep at least three version components to avoid
         # userland breakage (e.g. #742226, #745984).
-        self.abiname_version = re.sub(r'^(\d+\.\d+)(?=-|$)', r'\1.0',
-                                      self.version.linux_version)
+        self.abiname_version = self.version.linux_upstream_full
         self.vars = {
             'upstreamversion': self.version.linux_upstream,
             'version': self.version.linux_version,
