diff --color -Naur a/debian/rules.d/scripts/mod/Makefile.real b/debian/rules.d/scripts/mod/Makefile.real
--- a/debian/rules.d/scripts/mod/Makefile.real	2024-04-17 13:55:00.878122726 +0200
+++ b/debian/rules.d/scripts/mod/Makefile.real	2024-04-17 13:55:45.434133015 +0200
@@ -5,7 +5,7 @@
 wrapperdir = $(top_rulesdir)/$(OUTDIR)
 CFLAGS += -I $(CURDIR)/real-$(TYPE) -I $(wrapperdir)/real-$(TYPE)
 
-modpost.real-$(TYPE): file2alias.real-$(TYPE).o modpost.real-$(TYPE).o sumversion.real-$(TYPE).o
+modpost.real-$(TYPE): file2alias.real-$(TYPE).o modpost.real-$(TYPE).o sumversion.real-$(TYPE).o symsearch.real-$(TYPE).o
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^
 
 %.real-$(TYPE).o: $(SOURCEDIR)/%.c real-$(TYPE)/devicetable-offsets.h
