From ca3b6ec045153f7feb3584f3b9152c13c3ea1569 Mon Sep 17 00:00:00 2001
From: anandaravuri <ananda.ravuri@intel.com>
Date: Tue, 12 Dec 2023 11:05:57 -0800
Subject: [PATCH] fpga:update cxl cache buffer map flags

Signed-off-by: anandaravuri <ananda.ravuri@intel.com>
---
 include/uapi/linux/fpga-dfl.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/uapi/linux/fpga-dfl.h b/include/uapi/linux/fpga-dfl.h
index bf4c3223ce61..412c3c9e6c76 100644
--- a/include/uapi/linux/fpga-dfl.h
+++ b/include/uapi/linux/fpga-dfl.h
@@ -366,7 +366,7 @@ struct dfl_cxl_cache_region_info {
  */
 struct dfl_cxl_cache_buffer_map {
 	__u32 argsz;
-#define DFL_CXL_BUFFER_MAP_WRITABLE	BIT(0)
+#define DFL_CXL_BUFFER_MAP_WRITABLE	1
 	__u32 flags;
 	__u64 user_addr;
 	__u64 length;
