--- a/drivers/net/ethernet/intel/Kconfig	2023-02-28 14:27:47.919630051 +0100
+++ b/drivers/net/ethernet/intel/Kconfig	2023-02-28 14:28:23.847564543 +0100
@@ -375,6 +375,7 @@
 config S10HSSI
 	tristate "Control Plane Driver for Stratix 10 HSSI"
 	select REGMAP_INDIRECT_REGISTER
+	depends on FPGA_DFL
 	help
 	  This driver provides control plane support for an Stratix 10
 	  High Speed Serial Interface.

